platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/components/transaction/system/ConsensusSystemTransactionManager.java

```diff
@@ -36,42 +33,32 @@
 import org.apache.logging.log4j.Logger;
 
 /**
- * Implementation of {@link PostConsensusSystemTransactionManager}
+ * Routs consensus system transactions to the appropriate handlers.
  */
-public class PostConsensusSystemTransactionManagerImpl implements PostConsensusSystemTransactionManager {
+public class ConsensusSystemTransactionManager {
 
     /**
      * Class logger
      */
-    private static final Logger logger = LogManager.getLogger(PostConsensusSystemTransactionManagerImpl.class);
+    private static final Logger logger = LogManager.getLogger(ConsensusSystemTransactionManager.class);
 
     /**
      * The post-consensus handle methods that have been registered
      */
-    private final Map<
-                    Class<? extends SystemTransaction>, List<PostConsensusSystemTransactionHandler<SystemTransaction>>>
-            handlers = new HashMap<>();
+    private final Map<Class<?>, List<ConsensusSystemTransactionHandler<SystemTransaction>>> handlers = new HashMap<>();
 
     /**
      * Add a handle method
      *
-     * @param handler the new handler being tracked
+     * @param clazz   the class of the transaction being handled
+     * @param handler a method to handle this transaction type
      */
     @SuppressWarnings("unchecked")
-    private void addHandler(final PostConsensusSystemTransactionTypedHandler<?> handler) {
-        handlers.computeIfAbsent(handler.transactionClass(), k -> new ArrayList<>())
-                .add((PostConsensusSystemTransactionHandler<SystemTransaction>) handler.handleMethod());
-    }
+    public <T extends SystemTransaction> void addHandler(
+            @NonNull final Class<T> clazz, @NonNull final ConsensusSystemTransactionHandler<T> handler) {
```

Comment:
nit: in `PreconsensusSystemTransactionHandler::addHandler`, you null check `clazz` and `handler`. may as well have consistency between the classes

Reply:
nullity checks added